### PR TITLE
support increment type for cassandra and mongo

### PR DESCRIPTION
**Pull request by romanesko**

### CODE REVIEW

File: twake/backend/node/src/core/platform/services/database/services/orm/connectors/cassandra/cassandra.ts [hunk 1]
```diff
@@ -316,7 +316,11 @@ export class CassandraConnector extends AbstractConnector<
             `${transformValueToDbString(
               entity[columnsDefinition[key].nodename],
               columnsDefinition[key].type,
-              { columns: columnsDefinition[key].options, secret: this.secret },
+              {
+                columns: columnsDefinition[key].options,
+                secret: this.secret,
+                column: { key: key },
```
Comment (chamerling) [inline twake/backend/node/src/core/platform/services/database/services/orm/connectors/cassandra/cassandra.ts:322]:
```
column: { key }
```

is enough

File: twake/backend/node/src/core/platform/services/database/services/orm/connectors/cassandra/cassandra.ts [hunk 2]
```diff
@@ -329,6 +333,7 @@ export class CassandraConnector extends AbstractConnector<
               columns: columnsDefinition[key].options,
               secret: this.secret,
               disableSalts: true,
+              column: { key: key },
```
Comment (chamerling) [inline twake/backend/node/src/core/platform/services/database/services/orm/connectors/cassandra/cassandra.ts:336]: Same here and below

File: twake/backend/node/src/core/platform/services/database/services/orm/connectors/cassandra/typeTransforms.ts [hunk 3]
```diff
@@ -92,6 +92,10 @@ export const transformValueToDbString = (
     }
     return `'${(v || "").toString().replace(/'/gm, "''")}'`;
   }
+  if (type === "counter") {
+    if (isNaN(v)) throw Error("Counter value should be a number");
```
Comment (chamerling) [inline twake/backend/node/src/core/platform/services/database/services/orm/connectors/cassandra/typeTransforms.ts:96]: `throw new Error()`

File: twake/backend/node/src/core/platform/services/database/services/orm/connectors/cassandra/typeTransforms.ts [hunk 4]
```diff
@@ -154,7 +158,11 @@ export const transformValueFromDbString = (
   }
 
   if (type === "number") {
-    return new Number(v).valueOf();
+    return Number(v).valueOf();
```
Comment (chamerling) [inline twake/backend/node/src/core/platform/services/database/services/orm/connectors/cassandra/typeTransforms.ts:161]: is it better?

File: twake/backend/node/src/core/platform/services/database/services/orm/connectors/mongodb/typeTransforms.ts [hunk 5]
```diff
@@ -58,6 +58,11 @@ export const transformValueToDbString = (
     return Boolean(v);
   }
 
+  if (type === "counter") {
+    if (isNaN(v)) throw Error("Counter value should be a number");
```
Comment (chamerling) [inline twake/backend/node/src/core/platform/services/database/services/orm/connectors/mongodb/typeTransforms.ts:62]: new Error

File: twake/backend/node/test/e2e/counters.test.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    );
+
+    ends();
+  });
+
+  afterAll(done => {
+    platform.tearDown().then(done);
+  });
+
+  it("Initing empty value", async done => {
```
Comment (chamerling) [inline twake/backend/node/test/e2e/counters.test.ts:64]: `Initializing empty value`

**Review by chamerling — CHANGES_REQUESTED** [review]

**Review by RomaricMourgues — APPROVED** [review]

**Review by chamerling — APPROVED** [review]

**Pull request merged**